File: WinForm Demo (VB.NET)/MySqlBackupDemoVB/frmBLOB.vb
Imports MySql.Data.MySqlClient
Public Class frmBLOB

    Private constr As String = ""

    Public Sub New(ByVal connectionstring As String)

        InitializeComponent()

        constr = connectionstring

        Using conn As New MySqlConnection(constr)
            conn.Open()
            Dim dt As New DataTable()
            Dim sql As String = "show tables;"
            Dim da As New MySqlDataAdapter(sql, conn)
            da.Fill(dt)
            For Each dr As DataRow In dt.Rows
                cbTable.Items.Add(dr(0) + "")
            Next
            conn.Close()
        End Using

        cbTable.SelectedIndex = -1
        AddHandler cbTable.SelectedIndexChanged, AddressOf cbTable_SelectedIndexChanged

    End Sub

    Private Sub cbTable_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)

        Using conn As New MySqlConnection(constr)
            conn.Open()
            Dim dt As New DataTable()
            Dim sql As String = "show columns from `" + cbTable.Text + "`;"
            Dim da As New MySqlDataAdapter(sql, conn)
            da.Fill(dt)
            cbBLOB.Items.Clear()
            cbFileName.Items.Clear()
            cbFileSize.Items.Clear()
            Dim hasBLOB As Boolean = False
            For Each dr As DataRow In dt.Rows
                Dim colname As String = dr("Field") + ""
                Dim coltype As String = dr("Type") + ""
                If coltype.ToLower().Contains("blob") Then
                    cbBLOB.Items.Add(colname)
                    hasBLOB = True
                Else
                    cbFileName.Items.Add(colname)
                    cbFileSize.Items.Add(colname)
                End If
            Next
            conn.Close()
            If hasBLOB Then
                label5.Visible = False
            Else
                label5.Visible = True
            End If
        End Using

        Try
            cbBLOB.SelectedIndex = 0
            cbFileName.SelectedIndex = 0
            cbFileSize.SelectedIndex = 1
        Catch
        End Try

    End Sub

    Private Sub cmdExport_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cmdExport.Click
        Try
            Dim mb As New MySql.Data.MySqlClient.MySqlBackup(constr)
            mb.ExportBlobAsFile(txtFolder.Text, cbTable.Text, cbBLOB.Text, cbFileName.Text, cbFileSize.Text)
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
            Return
        End Try
        MessageBox.Show("Files have exported successfully and saved to" _
            & vbCr & vbLf & vbCr & vbLf & txtFolder.Text)
    End Sub

    Private Sub cmdFolder_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cmdFolder.Click
        Using f As New FolderBrowserDialog()
            f.ShowNewFolderButton = True
            If f.ShowDialog() = System.Windows.Forms.DialogResult.OK Then _
                txtFolder.Text = f.SelectedPath
        End Using
    End Sub

End Class

File: WinForm Demo (C#)/MySqlBackupDemo/frmBLOB.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using MySql.Data.MySqlClient;

namespace MySqlBackupDemo
{
    public partial class frmBLOB : Form
    {
        string constr = "";

        public frmBLOB(string connectionstring)
        {
            InitializeComponent();
            constr = connectionstring;

            using (MySqlConnection conn = new MySqlConnection(constr))
            {
                conn.Open();
                DataTable dt = new DataTable();
                string sql = "show tables;";
                MySqlDataAdapter da = new MySqlDataAdapter(sql, conn);
                da.Fill(dt);
                foreach(DataRow dr in dt.Rows)
                {
                    cbTable.Items.Add(dr[0] + "");
                }
                conn.Close();
            }
            cbTable.SelectedIndex = -1;
            cbTable.SelectedIndexChanged += new EventHandler(cbTable_SelectedIndexChanged);
        }

        void cbTable_SelectedIndexChanged(object sender, EventArgs e)
        {
            using (MySqlConnection conn = new MySqlConnection(constr))
            {
                conn.Open();
                DataTable dt = new DataTable();
                string sql = "show columns from `" + cbTable.Text + "`;";
                MySqlDataAdapter da = new MySqlDataAdapter(sql, conn);
                da.Fill(dt);
                cbBLOB.Items.Clear();
                cbFileName.Items.Clear();
                cbFileSize.Items.Clear();
                bool hasBLOB = false;
                foreach (DataRow dr in dt.Rows)
                {
                    string colname = dr["Field"] + "";
                    string coltype = dr["Type"] + "";
                    if (coltype.ToLower().Contains("blob"))
                    {
                        cbBLOB.Items.Add(colname);
                        hasBLOB = true;
                    }
                    else
                    {
                        cbFileName.Items.Add(colname);
                        cbFileSize.Items.Add(colname);
                    }
                }
                conn.Close();
                if (hasBLOB)
                    label5.Visible = false;
                else
                    label5.Visible = true;
            }
            try
            {
                cbBLOB.SelectedIndex = 0;
                cbFileName.SelectedIndex = 0;
                cbFileSize.SelectedIndex = 1;
            }
            catch { }
        }

        private void cmdExport_Click(object sender, EventArgs e)
        {
            try
            {
                MySql.Data.MySqlClient.MySqlBackup mb = new MySql.Data.MySqlClient.MySqlBackup(constr);
                mb.ExportBlobAsFile(txtFolder.Text, cbTable.Text, cbBLOB.Text, cbFileName.Text, cbFileSize.Text);
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
                return;
            }
            MessageBox.Show("Files have exported successfully and saved to\r\n\r\n" + txtFolder.Text);
        }

        private void cmdFolder_Click(object sender, EventArgs e)
        {
            FolderBrowserDialog f = new FolderBrowserDialog();
            f.ShowNewFolderButton = true;
            if (f.ShowDialog() == System.Windows.Forms.DialogResult.OK)
            {
                txtFolder.Text = f.SelectedPath;
            }
        }
    }
}
